pkg/operator/operator.go

```diff
@@ -0,0 +1,279 @@
+package operator
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	appsclientv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	operatorsv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	scsclientv1alpha1 "github.com/openshift/client-go/servicecertsigner/clientset/versioned/typed/servicecertsigner/v1alpha1"
+	scsinformerv1alpha1 "github.com/openshift/client-go/servicecertsigner/informers/externalversions/servicecertsigner/v1alpha1"
+	"github.com/openshift/service-serving-cert-signer/pkg/forlibrarygo/v1alpha1helpers"
+)
+
+const (
+	targetNamespaceName = "openshift-service-cert-signer"
+	workQueueKey        = "key"
+)
+
+type ServiceCertSignerOperator struct {
+	operatorConfigClient scsclientv1alpha1.ServiceCertSignerOperatorConfigsGetter
+
+	appsv1Client appsclientv1.AppsV1Interface
+	corev1Client coreclientv1.CoreV1Interface
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewServiceCertSignerOperator(
+	serviceCertSignerConfigInformer scsinformerv1alpha1.ServiceCertSignerOperatorConfigInformer,
+	namespacedKubeInformers informers.SharedInformerFactory,
+	operatorConfigClient scsclientv1alpha1.ServiceCertSignerOperatorConfigsGetter,
+	appsv1Client appsclientv1.AppsV1Interface,
+	corev1Client coreclientv1.CoreV1Interface,
+) *ServiceCertSignerOperator {
+	c := &ServiceCertSignerOperator{
+		operatorConfigClient: operatorConfigClient,
+		appsv1Client:         appsv1Client,
+		corev1Client:         corev1Client,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "WebConsoleOperator"),
```

Comment:
s/WebConsoleOperator/ServiceServingCertSignerOperator/ (that name is horrible btw ;-)